lib/utils.js

```diff
@@ -538,6 +538,23 @@ exports.populate = function populate(path, select, model, match, options, subPop
     throw new TypeError('utils.populate: invalid path. Expected string. Got typeof `' + typeof path + '`');
   }
 
+  if (Array.isArray(subPopulate)) {
+    let ret = [];
+    subPopulate.forEach(function(obj) {
+      if (/[\s]/.test(obj.path)) {
+        let copy = exports.object.shallowCopy(obj);
+        let paths = copy.path.split(' ');
+        paths.forEach(function(p) {
+          copy.path = p;
+          ret.push(exports.populate(copy)[0]);
+        });
+      } else {
+        ret.push(exports.populate(obj)[0]);
+      }
+    });
+    subPopulate = exports.populate(ret);
+  }
+
   if (typeof subPopulate === 'object') {
```

Comment:
Make this an `else`, because `typeof [] === 'object'` so this `if` will also execute.

Reply:
fixed.